GetStream/stream-chat-swift
# Fix not able to mark channel read after clearing history

### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/621

### 🎯 Goal
Fix not able to mark channel read after clearing the messages history.

### 🛠 Implementation
After clearing the messages history, we won't be able to find the `LastReadMessageId` in the `channelController.messages`, so the `firstUnreadMessageId` will always return nil. In order to fix this edge case, we need to check if we already fetched all messages, if yes, then it means the `firstUnreadMessage` is the top message of the list.

### 🎨 Showcase

The Device on the Left showcases the problem, the Device on the Right has the problem solved.

https://github.com/GetStream/stream-chat-swift/assets/12814114/dc19f6be-80c5-432f-b449-8b3abb6def16


### 🧪 Manual Testing Notes
1. Open the Channel List
2. Receive a message while in the Channel List
3. Hide and clear the history of that channel (Do not open the channel)
4. Receive again a new message on that same channel
5. The channel should re-appear on the Channel List
6. Open the channel
7. Go back to the Channel List
8. The channel should have the unreads count 0

### ☑️ Contributor Checklist
